### PR TITLE
feat: add the e2e test job for Cloud Build

### PR DESCRIPTION
**Pull request by hessjcg**
The Cloud Build configuration and scripts needed to run end-to-end tests. 

In this change, the job works, but must be triggered manually. We will add automatic triggering
for this job in future PRs.

Related to: #65

### CODE REVIEW

File: tools/e2e_test_job.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# Set the e2e test project id and other params from
+# the Cloud Build environment
+cat > build.env <<EOF
+E2E_PROJECT_ID=$E2E_PROJECT_ID
+NODEPOOL_SERVICEACCOUNT_EMAIL=$NODEPOOL_SERVICEACCOUNT_EMAIL
+WORKLOAD_ID_SERVICEACCOUNT_EMAIL=$WORKLOAD_ID_SERVICEACCOUNT_EMAIL
+TFSTATE_STORAGE_BUCKET=$TFSTATE_STORAGE_BUCKET
+EOF
+
+echo "Running tests on landscape ${ENVIRONMENT_NAME:-undefined}"
```
Comment (enocom) [inline tools/e2e_test_job.sh:60]: s/landscape/environment/

File: tools/e2e_test_job.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# to prepare the Cloud Build environment and run the end-to-end tests.
+#
+
+set -euxo
+
+E2E_PROJECT_ID=cloud-sql-operator-testing
+
+# Install GCC and other essential build tools
+apt-get update
+apt-get install -y zip unzip build-essential
```
Comment (enocom) [inline tools/e2e_test_job.sh:27]: How can we cache these steps such that we're not re-installed all these deps (and the ones below) for each e2e test build?

Comment (hessjcg) [reply tools/e2e_test_job.sh:27]: I had the same question so I measured it. 

- Installing tools: 2 minutes
- Reconciling cloud resources with Terraform (no changes required): 3 minutes
- Building and deploying the operator image: 12 minutes
- Running e2e tests: 5 minutes. 

Overall: 21 minutes

Building and deploying is too slow. Mainly two steps make it slow: 

- Downloading all the go libraries used by the operator and the operator's build process. And it happens twice: once in the build container `make generate` and once again during `docker buildx build.` 
- Running the container build using Docker-in-docker. This is a lot slower than a native go compiler. 

I think we need to optimize this, but in a different PR. For now, I'd like to call this working for now, but not yet fast enough.

File: tools/e2e_test_job.sh (hunk 1 above)
Comment (hessjcg) [reply tools/e2e_test_job.sh]: Good catch. Fixed.

File: tools/e2e_test_job.sh (hunk 2 above)
Comment (enocom) [reply tools/e2e_test_job.sh:28]: So basically every build of e2e is at least 21 minutes?

Comment (hessjcg) [reply tools/e2e_test_job.sh:28]: For now, yes.

**Review by enocom — APPROVED** [review]

**Pull request merged**